### PR TITLE
SNOW v2 - Added the *additional_fields* argument to the following commands:

### PR DESCRIPTION
**Pull request by bakatzir**
- ***servicenow-get-ticket***
    - ***servicenow-update-ticket***
    - ***servicenow-create-ticket***
    - ***servicenow-query-tickets***

## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/23525

## Screenshots
in the issue

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by bakatzir** [comment]
referenced: https://github.com/demisto/etc/issues/23708
@kirbles19 Please review the entire yml of this integration:
https://github.com/demisto/content/blob/d89bf7afe459823abb09261dc46d1655987b24c5/Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml
As it was yet reviewed by tech docs.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 1]
```diff
@@ -667,12 +676,14 @@ def get_ticket_command(client: Client, args: dict):
...
-    hr = get_ticket_human_readable(ticket, ticket_type)
-    context = get_ticket_context(ticket)
+    hr = get_ticket_human_readable(ticket, ticket_type, additional_fields)
+    context = get_ticket_context(ticket, additional_fields)
 
     headers = ['System ID', 'Number', 'Impact', 'Urgency', 'Severity', 'Priority', 'State', 'Approval',
                'Created On', 'Created By', 'Active', 'Close Notes', 'Close Code', 'Description', 'Opened At',
                'Due Date', 'Resolved By', 'Resolved At', 'SLA Due', 'Short Description', 'Additional Comments']
+    if additional_fields:
+        headers += additional_fields
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:686]: I don't think this works, use `extend` instead

**Comment by bakatzir** [comment]
@liorblob 
> We also want to be able to update/fill these fields in the `update` and `create` commands (in format of `arg=value`)

We have that using the `custom_fields` argument. in both get/update/create

**Comment by liorblob** [comment]
@bakatzir `custom_fields` are only for fields that start with `u_`. there are other fields as well...

**Comment by bakatzir** [comment]
@liorblob made the `additional_fields` to receive `arg=val` separated with `;`, without prefixing it with `u_` to allow updating and creating any existing system fields.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml [hunk 2]
```diff
@@ -672,6 +679,12 @@ script:
       name: approval
       required: false
       secret: false
+    - default: false
+      description: 'Fields in the format: fieldname1=value;fieldname2=value;'
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:683]:
```suggestion
      description: 'Additional fields in the format: fieldname1=value;fieldname2=value;'
```

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml [hunk 3]
```diff
@@ -1200,6 +1213,12 @@ script:
...
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.yml:1217]:
```suggestion
      description: 'Additional fields in the format: fieldname1=value;fieldname2=value;'
```

**Comment by bakatzir** [comment]
@liorblob README is currently completely lacking, we are on it, but not necessarily in this PR.

**Comment by bakatzir** [comment]
SNOW does not throw an error, but the field is not updated.

**Comment by bakatzir** [comment]
@yuvalbenshalom added an erorr on the `split_fields` util, and a UT.
Screen Shot 2020-04-22 at 17 06 02

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 4]
```diff
@@ -96,11 +96,12 @@ def get_server_url(server_url: str) -> str:
     return url
 
 
-def create_ticket_context(data: dict) -> Any:
+def create_ticket_context(data: dict, additional_fields: list = []) -> Any:
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:99]: I don't think you can do that, it should be `None` in the decleration.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 5]
```diff
@@ -140,30 +142,32 @@ def create_ticket_context(data: dict) -> Any:
     return createContext(context, removeNull=True)
 
 
-def get_ticket_context(data: Any) -> Any:
+def get_ticket_context(data: Any, additional_fields: list = []) -> Any:
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:145]: Same here

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 6]
```diff
@@ -140,30 +142,32 @@ def create_ticket_context(data: dict) -> Any:
...
 
     tickets = []
     for d in data:
-        tickets.append(create_ticket_context(d))
+        tickets.append(create_ticket_context(d, additional_fields))
     return tickets
 
 
-def get_ticket_human_readable(tickets, ticket_type: str) -> list:
+def get_ticket_human_readable(tickets, ticket_type: str, additional_fields: list = []) -> list:
```
Comment (liorblob) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:164]: And here

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 4 above)
Comment (liorblob) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:99]: image

**Pull request merged**